core/src/data_sources/data_source.rs

```diff
@@ -402,12 +402,30 @@ pub struct DocumentVersion {
     pub hash: String,
 }
 
+#[derive(Serialize, Deserialize, PartialEq, Clone, Debug)]
+pub struct EmbedderConfig {
+    pub provider_id: ProviderID,
+    pub model_id: String,
+    pub splitter_id: SplitterID,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Clone, Debug)]
+pub struct EmbedderDataSourceConfig {
+    pub embedder: EmbedderConfig,
+    pub shadow_embedder: Option<EmbedderConfig>,
+}
+
 #[derive(Serialize, Deserialize, PartialEq, Clone, Debug)]
 pub struct DataSourceConfig {
+    // TODO(2024-05-24 flav) Remove once migrated to embedder.
     pub provider_id: ProviderID,
     pub model_id: String,
-    pub extras: Option<Value>,
     pub splitter_id: SplitterID,
+
+    // Optional until we update api calls to pass this body.
+    pub embedder_config: Option<EmbedderDataSourceConfig>,
+
+    pub extras: Option<Value>,
     pub max_chunk_size: usize,
```

Comment:
We def want to move max_chunk_size as well as bumping chunk_size will require similar re-embedding :+1:

Reply:
🤔 We chat live about this on Friday and you said no. Anyway, I'll update.

Reply:
I was wrong I think :)